libr/cons/cons.c

```diff
@@ -44,6 +44,58 @@ static void cons_stack_free(void *ptr) {
 	free (s);
 }
 
+static RConsStack *cons_stack_dump(bool recreate) {
+	RConsStack *data = R_NEW0 (RConsStack);
+	if (!data) {
+		return NULL;
+	}
+
+	if (I.buffer) {
+		data->buf = I.buffer;
+		data->buf_len = I.buffer_len;
+		data->buf_size = I.buffer_sz;
+	}
+
+	data->grep = R_NEW0 (RConsGrep);
+	if (data->grep) {
+		memcpy (data->grep, &I.grep, sizeof (RConsGrep));
+		if (I.grep.str) {
+			data->grep->str = strdup (I.grep.str);
+		}
+	}
+
+	if (recreate) {
+		I.buffer = malloc (I.buffer_sz);
+		if (!I.buffer) {
+			I.buffer = data->buf;
+			free (data);
+			return NULL;
+		}
+	}
+
+	return data;
+}
+
+static void cons_stack_load(RConsStack *data, bool free_current) {
+	if (free_current) {
+		free (I.buffer);
+	}
+
+	I.buffer = data->buf;
+	data->buf = NULL;
+	I.buffer_len = data->buf_len;
+	I.buffer_sz = data->buf_size;
+
+	if (data->grep) {
+		memcpy (&I.grep, data->grep, sizeof (RConsGrep));
+		if (data->grep->str) {
+			char *old = I.grep.str;
+			I.grep.str = strdup (data->grep->str);
+			R_FREE (old);
+		}
```

Comment:
i dont see the point on keeping that copy and then freeing it with null

Reply:
Not my code, I just moved it. But yeah, looks weird.

Reply:
It comes from this commit: https://github.com/radare/radare2/commit/d4ee6f6e955057d412b01221289ea39272559cc1